### PR TITLE
Patch pods to add labels or annotations, rather than restarting

### PR DESCRIPTION
**Pull request by russgold**
When a pod is missing a label or annotation specified in the domain, we no longer recycle it; instead, we patch it. 

We should probably do the same for services, but there is some cleanup that will be needed to prepare for that.

### CODE REVIEW

File: operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java [hunk 1]
```diff
@@ -472,6 +529,8 @@ public NextAction apply(Packet packet) {
       V1Pod currentPod = getSko().getPod().get();
       if (currentPod == null) {
         return doNext(createNewPod(getNext()), packet);
+      } else if (mustPatchPod(currentPod)) {
```
Comment (rjeberhard) [inline operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java]: Question on the intention here (and we might need another test)...  What if the domain resource has changed such that there would both be an additional label and an additional volume?  For that use case, I believe that the pod needs to be replaced, but I think the else clause ordering would just patch the label.

File: site/server-lifecycle.md [hunk 2]
```diff
@@ -132,7 +132,6 @@ The operator runtime does rolling restarts of clustered servers so that service
 
 ### Properties that cause servers to be restarted
 The operator will restart servers when any of the follow properties on the domain resource that affect the server are changed:
-* `annotations`
```
Comment (rjeberhard) [inline site/server-lifecycle.md]: Any thoughts on how we should document that servers may have labels or annotations that were previously specified on the domain resource given that the server won't be restarted?

File: operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java (hunk 1 above)
Comment (russgold) [reply operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java]: I've now changed the order of conditionals to handle that.

File: site/server-lifecycle.md (hunk 2 above)
Comment (russgold) [reply site/server-lifecycle.md]: I've added a note.

**Pull request merged**